versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/logic/RepositoryLogicImpl.java

```diff
@@ -169,14 +184,51 @@ private void addRepositoryDescription(
                   null,
                   "application/json",
                   SHARED_OBJECT_MAPPER.writeValueAsBytes(repoDesc.build()));
-      // can safely ignore the ID returned from storeObj() - it's fine, if the obj already exists
+      // can safely ignore the response from storeObj() - it's fine, if the obj already exists
       persist.storeObj(string);
-      b.addAdds(commitAdd(KEY_REPO_DESCRIPTION, 0, requireNonNull(string.id()), null, null));
+      b.addAdds(
+          commitAdd(KEY_REPO_DESCRIPTION, 0, requireNonNull(string.id()), null, UUID.randomUUID()));
     } catch (ObjTooLargeException | ObjNotFoundException | IOException e) {
       throw new RuntimeException(e);
     }
   }
 
+  @Override
+  public RepositoryDescription updateRepositoryDescription(
+      RepositoryDescription repositoryDescription) throws RetryTimeoutException {
+    try {
+      return commitRetry(
+          persist,
+          (p, retryState) -> {
+            try {
+              Reference reference = Objects.requireNonNull(persist.fetchReference(REF_REPO.name()));
```

Comment:
We should probably fix the javadoc of `REF_REPO`, which currently says `Internal reference with always exactly one commit ...`